### PR TITLE
ARACHNE-2729 Add IAP auth support into Standalone Node

### PR DESCRIPTION
**Pull request by YaroslavTir**
ARACHNE-2729  Add IAP auth support into Standalone Node

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -321,7 +321,7 @@
         <dependency>
             <groupId>org.ohdsi</groupId>
             <artifactId>authenticator</artifactId>
-            <version>0.0.1-QA</version>
+            <version>0.0.2-SNAPSHOT</version>
```
Comment (acumarav) [inline pom.xml]: we probably should release 0.0.2-qa and refer it here.

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java [hunk 2]
```diff
@@ -102,11 +106,13 @@
     @RequestMapping(value = "/api/v1/auth/method", method = GET)
     @Deprecated
     public JsonResult<CommonAuthMethodDTO> authMethod() {
-
-        if (!FunctionalMode.NETWORK.equals(dataNodeService.getDataNodeMode())) {
-            throw new BadRequestException();
+        if (dataNodeService.getDataNodeMode() == FunctionalMode.STANDALONE) {
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: > Even couldn't compile the module:
> 
> ```
> Error:(137, 67) java: cannot find symbol
>   symbol:   method getTokenHeader()
>   location: variable accessTokenResolver of type org.ohdsi.authenticator.service.AccessTokenResolver
> ```

This is part of the authenticator library, and this one is not merged yet

[the remark this answers is not in the document]
Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/controller/AuthController.java]: Indeed, but am I right this should complain to Authenticator#ARACHNE-2729-IAP-auth-support branch?
And I see the `accessTokenResolver.getTokenHeader()` in the DataNode
And `AccessTokenResolver.getTokenHeaderName` in the Authenticator
What I'm doing wrong?

File: src/main/java/com/odysseusinc/arachne/datanode/service/client/portal/CentralRequestInterceptor.java [hunk 3]
```diff
@@ -53,8 +53,9 @@ public CentralRequestInterceptor(ApplicationContext applicationContext) {
     private String getToken() {
 
         final Object credentials = SecurityContextHolder.getContext().getAuthentication().getCredentials();
-        if (credentials instanceof String && StringUtils.isNotBlank((String)credentials)) {
-            String centralToken = tokenService.resolveAdditionalInfo(credentials.toString(), "token", String.class);
+        if (credentials instanceof String) {
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/client/portal/CentralRequestInterceptor.java]: I am kind of confused, If I open the file directly this check is there:
https://github.com/OHDSI/ArachneNodeAPI/blob/ARACHNE-2729-IAP-auth-support/src/main/java/com/odysseusinc/arachne/datanode/service/client/portal/CentralRequestInterceptor.java

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -74,4 +86,26 @@ public void afterPropertiesSet() throws Exception {
 
         userDetailsService = applicationContext.getBean(UserDetailsService.class);
     }
+
+    private void createUserByTokenIfNecessary(String username) {
+        if (authenticationMode == AuthenticationMode.STANDARD) {
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/AuthenticationServiceImpl.java]: fixed

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -74,4 +86,26 @@ public void afterPropertiesSet() throws Exception {
...
+        }
+        if (StringUtils.isEmpty(username)) {
+            throw new AuthException("Username cannot be empty.");
+        }
+        User user = getUserByUserEmail(username);
+        userRegisterStrategy.registerUser(user);
+    }
+
+    private User getUserByUserEmail(String email) {
+        String name = StringUtils.split(email, "@")[0];
```
[the remark this answers is not in the document]
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/AuthenticationServiceImpl.java]: fixed

**Pull request merged**